src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/ParenthesizedExpressionSyntaxExtensions.cs

```diff
@@ -26,6 +26,12 @@ public static bool CanRemoveParentheses(this ParenthesizedExpressionSyntax node,
             }
 
             var expression = node.Expression;
+            if (expression is StackAllocArrayCreationExpressionSyntax)
+            {
+                // var span = (stackalloc byte[8]);
+                // https://github.com/dotnet/roslyn/issues/44629
```

Comment:
it woudl be a worth a dedicated comment for this.  also, this check should likely go lower.  i.e. if we have `((stackalloc byte[8]))` we'd want to remove the outer parens.

Reply:
@CyrusNajmabadi, 
> it woudl be a worth a dedicated comment for this.
Do you mean a comment linking the PR?

> i.e. if we have ((stackalloc byte[8])) we'd want to remove the outer parens.
Right, I'll move it down.

Reply:
The current build fails without enough information available to me.

```
Compilers.sln:
/Users/runner/runners/2.169.1/work/1/s/.dotnet/sdk/3.1.200/MSBuild.dll /nologo -maxcpucount /m -verbosity:m /v:minimal /bl:/Users/runner/runners/2.169.1/work/1/s/artifacts/log/Debug/ToolsetRestore.binlog /clp:Summary /clp:ErrorsOnly;NoSummary /nr:false /p:TreatWarningsAsErrors=false /p:ContinuousIntegrationBuild=true /p:__ToolsetLocationOutputFile=/Users/runner/runners/2.169.1/work/1/s/artifacts/toolset/1.0.0-beta.20278.5.txt /t:__WriteToolsetLocation /Users/runner/runners/2.169.1/work/1/s/artifacts/toolset/restore.proj
/Users/runner/runners/2.169.1/work/1/s/artifacts/toolset/restore.proj : error : System.AggregateException: One or more errors occurred. (Could not find a part of the path '/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/NuGetScratch/0709bfa4352c4b61a70bc6fb18e24971/a9376905db214e218fcd506fe4571a66.proj.nuget.dgspec.json'.)
/Users/runner/runners/2.169.1/work/1/s/artifacts/toolset/restore.proj : error :  ---> System.IO.DirectoryNotFoundException: Could not find a part of the path '/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/NuGetScratch/0709bfa4352c4b61a70bc6fb18e24971/a9376905db214e218fcd506fe4571a66.proj.nuget.dgspec.json'.
/Users/runner/runners/2.169.1/work/1/s/artifacts/toolset/restore.proj : error :    at Interop.ThrowExceptionForIoErrno(ErrorInfo errorInfo, String path, Boolean isDirectory, Func`2 errorRewriter)
/Users/runner/runners/2.169.1/work/1/s/artifacts/toolset/restore.proj : error :    at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String path, OpenFlags flags, Int32 mode)
/Users/runner/runners/2.169.1/work/1/s/artifacts/toolset/restore.proj : error :    at System.IO.FileStream.OpenHandle(FileMode mode, FileShare share, FileOptions options)
/Users/runner/runners/2.169.1/work/1/s/artifacts/toolset/restore.proj : error :    at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
/Users/runner/runners/2.169.1/work/1/s/artifacts/toolset/restore.proj : error :    at System.IO.FileStream..ctor(String path, FileMode mode)
/Users/runner/runners/2.169.1/work/1/s/artifacts/toolset/restore.proj : error :    at NuGet.ProjectModel.DependencyGraphSpec.Save(String path)
/Users/runner/runners/2.169.1/work/1/s/artifacts/toolset/restore.proj : error :    at NuGet.Commands.NoOpRestoreUtilities.PersistDGSpecFile(DependencyGraphSpec spec, String dgPath, ILogger log)
/Users/runner/runners/2.169.1/work/1/s/artifacts/toolset/restore.proj : error :    at NuGet.Commands.RestoreCommand.EvaluateCacheFile()
/Users/runner/runners/2.169.1/work/1/s/artifacts/toolset/restore.proj : error :    at NuGet.Commands.RestoreCommand.ExecuteAsync(CancellationToken token)
/Users/runner/runners/2.169.1/work/1/s/artifacts/toolset/restore.proj : error :    at NuGet.Commands.RestoreRunner.ExecuteAsync(RestoreSummaryRequest summaryRequest, CancellationToken token)
/Users/runner/runners/2.169.1/work/1/s/artifacts/toolset/restore.proj : error :    at NuGet.Commands.RestoreRunner.CompleteTaskAsync(List`1 restoreTasks)
/Users/runner/runners/2.169.1/work/1/s/artifacts/toolset/restore.proj : error :    at NuGet.Commands.RestoreRunner.RunWithoutCommit(IEnumerable`1 restoreRequests, RestoreArgs restoreContext)
/Users/runner/runners/2.169.1/work/1/s/artifacts/toolset/restore.proj : error :    --- End of inner exception stack trace ---
/Users/runner/runners/2.169.1/work/1/s/artifacts/toolset/restore.proj : error :    at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
/Users/runner/runners/2.169.1/work/1/s/artifacts/toolset/restore.proj : error :    at System.Threading.Tasks.Task`1.get_Result()
/Users/runner/runners/2.169.1/work/1/s/artifacts/toolset/restore.proj : error :    at Microsoft.Build.NuGetSdkResolver.NuGetSdkResolver.NuGetAbstraction.GetSdkResult(SdkReference sdk, Object nuGetVersion, SdkResolverContext context, SdkResultFactory factory)
/Users/runner/runners/2.169.1/work/1/s/artifacts/toolset/restore.proj : error MSB4236: The SDK 'Microsoft.DotNet.Arcade.Sdk' specified could not be found.
##[error]Build failed (exit code '1').
Killing running build processes...
```

Is that related to my changes or something else?

Reply:
ignore that.  ci machine issue.